external/llvm-project/mlir/lib/Dialect/Tosa/Transforms/TosaPartition.cpp

```diff
@@ -475,7 +477,8 @@ void TosaPartitionPass::traceInputs(Operation *op,
         // depth first
         traceInputs(usedOp, predecessors, inputNodes);
       }
-    } else {
+    } else if (!predecessors.contains(
+                   usedOp)) { // special-case consts aren't inputs
```

Comment:
This change and the one below avoid adding a node to inputNodes if it's already present in the saved nodes.  That generalises the problem of the special-case transpose constants -- since they don't qualify to be saved under normal circumstances, they'd otherwise fall into the inputNodes case, but we don't want to do that because it would just be an extra unused argument to the outlined function.  Cosmetic, but it bugged me.